strawberry/subscriptions/protocols/graphql_transport_ws/handlers.py

```diff
@@ -164,6 +167,8 @@ async def handle_subscribe(self, message: SubscribeMessage) -> None:
             )
 
         context = await self.get_context()
+        if isinstance(context, dict):
```

Comment:
I'm still thinking about this part. It means the base class would only add the connection params to the context if the context is a dict. In case for at least one integration (fastapi) that would not be the case. That's why you handled it seperately there. I'm wondering whether we should let all the integrations handle it by themselves for consistency or whether we have a way to make it work with any context (tricky). Still not sure. What do you think? :thinking:

Reply:
My thoughts around this were to achieve as much generality as possible, while still maintaining specificity. I think that Strawberry should be able to handle subscriptions in the absence of an integration, e.g. by using the CLI, so I think `connection_params` must be injected here.

Alternatively, we could attempt to implement a general Strawberry-wide class-based context object, similar to the FastAPI one, and then have these two options be the standard implementations that integrations should sub-class from. Indeed, I remember @jkimbo opening such an issue here: https://github.com/strawberry-graphql/strawberry/issues/1258.

Reply:
> achieve as much generality as possible

Good point, I was only concerned about maintainability. But I think it's okay :)

> e.g. by using the CLI

The CLI uses the ASGI integration under the hood.

> Alternatively, we could attempt to implement a general Strawberry-wide class-based context object

I'm very much in favour of that. But that would be another PR of course :)